wallet/wallet.go

```diff
@@ -4,70 +4,358 @@ package wallet
+func (w *LocalWallet) loadAccountIntoWallet(ctx context.Context, account account.Account, network network.Network) error {
 	wallet, err := primary.MakeWallet(
 		ctx,
-		uri,
-		avaxKeychain,
-		nil,
-		config,
+		network.Endpoint,
+		account.SoftKey.KeyChain(),
+		account.SoftKey.KeyChain(),
+		primary.WalletConfig{},
 	)
-	return Wallet{
-		Wallet: wallet,
-		Keychain: keychain.Keychain{
-			Keychain: avaxKeychain,
-		},
-		config: config,
-	}, err
-}
-
-// SecureWalletIsChangeOwner ensures that a fee paying address (wallet's keychain) will receive
-// the change UTXO and not a randomly selected auth key that may not be paying fees
-func (w *Wallet) SecureWalletIsChangeOwner() {
-	addrs := w.Addresses()
-	changeAddr := addrs[0]
-	// sets change to go to wallet addr (instead of any other subnet auth key)
-	changeOwner := &secp256k1fx.OutputOwners{
-		Threshold: 1,
-		Addrs:     []ids.ShortID{changeAddr},
+	if err != nil {
+		return err
+	}
+	w.Wallet = wallet
+	return nil
+}
+
+func (w *LocalWallet) Accounts() []account.Account {
+	return w.accounts
+}
+
+func (w *LocalWallet) Clients() ChainClients {
+	return w.clients
+}
+
+// CreateAccount creates a new Account using local key generation
+func (w *LocalWallet) CreateAccount(ctx context.Context) (*account.Account, error) {
+	newAccount, err := account.NewAccount()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new Account: %w", err)
+	}
+
+	// Add the Account to the wallet
+	w.AddAccount(newAccount)
+
+	return &newAccount, nil
+}
+
+// GetAccount retrieves an existing Account by address or identifier
+func (w *LocalWallet) GetAccount(ctx context.Context, address ids.ShortID) (*account.Account, error) {
+	// TODO: Implement Account retrieval logic based on address
+	// This could search through w.accounts or use the embedded primary.Wallet
+	return nil, fmt.Errorf("not implemented")
+}
+
+// ListAccounts returns all accounts managed by this wallet
+func (w *LocalWallet) ListAccounts(ctx context.Context) ([]*account.Account, error) {
+	// Return all accounts in the wallet
+	accounts := w.GetAllAccounts()
+	result := make([]*account.Account, len(accounts))
+	for i := range accounts {
+		result[i] = &accounts[i]
+	}
+	return result, nil
+}
+
+// ImportAccount imports an existing Account into the wallet
+func (w *LocalWallet) ImportAccount(ctx context.Context, keyPath string) (*account.Account, error) {
+	// TODO: Implement Account import logic
+	// This would add the provided Account to the wallet
+	existingAccount, err := account.Import(keyPath)
+	if err != nil {
+		return nil, fmt.Errorf("error when importing Account %s \n", err)
+	}
+	w.AddAccount(existingAccount)
+	return &existingAccount, nil
+}
+
+// BuildTx constructs a transaction for the specified operation
+func (w *LocalWallet) BuildTx(ctx context.Context, params BuildTxParams) (tx.BuildTxResult, error) {
+	if err := w.loadAccountIntoWallet(ctx, params.Account, params.Network); err != nil {
```

Comment:
Is loadAccountIntoWallet idempotent?

Reply:
loadaccountintowallet clears any prexisting account in the wallet and load the specified account into the wallet.